gateway/src/shard/tls.rs

```diff
@@ -0,0 +1,161 @@
+//! TLS manager to reuse connections between shards.
+
+#[cfg(feature = "rustls")]
+use std::sync::Arc;
+use std::{
+    error::Error,
+    fmt::{Display, Formatter, Result as FmtResult},
+};
+
+#[cfg(all(feature = "native", not(feature = "rustls")))]
+use native_tls::TlsConnector as NativeTlsConnector;
+#[cfg(feature = "rustls")]
+use rustls_tls::ClientConfig;
+#[cfg(all(feature = "rustls", feature = "rustls-webpki-roots"))]
+use rustls_tls::OwnedTrustAnchor;
+use tokio_tungstenite::Connector;
```

Comment:
this is also not affected by the directive above, if it is meant to

Reply:
They are not, I have added some more spacing to make it clear